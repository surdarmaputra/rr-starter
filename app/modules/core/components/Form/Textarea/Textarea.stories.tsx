import type { Meta, StoryObj } from '@storybook/react';

import { Textarea, type TextareaProps } from './Textarea';

type Story = StoryObj<typeof Textarea>;

const meta: Meta = {
  component: Textarea,
  title: 'Core/Components/Form/Textarea',
};

export default meta;

export const Default: Story = {
  render: (args: TextareaProps) => <Textarea {...args} />,
};
