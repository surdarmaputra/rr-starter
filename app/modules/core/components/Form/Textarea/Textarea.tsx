import * as React from 'react';

import { mergeClassNames } from '~/modules/core/libs/utils/ui';

export type TextareaProps = React.ComponentPropsWithoutRef<'textarea'>;

const Textarea = React.forwardRef<HTMLTextAreaElement, TextareaProps>(
  ({ className, ...props }, ref) => {
    return (
      <textarea
        className={mergeClassNames(
          'flex min-h-[80px] w-full rounded-md border border-divider bg-white px-3 py-2 text-base ring-offset-white',
          'placeholder:text-letter-caption focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-primary focus-visible:ring-offset-1',
          'disabled:cursor-not-allowed disabled:opacity-50 md:text-sm',
          className,
        )}
        ref={ref}
        {...props}
      />
    );
  },
);
Textarea.displayName = 'Textarea';

export { Textarea };
