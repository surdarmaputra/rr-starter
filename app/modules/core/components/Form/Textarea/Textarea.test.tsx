import { render, screen } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import { Textarea, type TextareaProps } from './Textarea';

describe('Textarea', () => {
  function renderComponent(props?: TextareaProps) {
    return render(<Textarea data-testid="Textarea-testid" {...props} />);
  }

  test('renders with data-testid', async () => {
    renderComponent();
    await screen.findByTestId('Textarea-testid');
  });

  test('renders snapshot', async () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
