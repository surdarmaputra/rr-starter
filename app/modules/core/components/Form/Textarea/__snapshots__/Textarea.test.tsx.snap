// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Textarea > renders snapshot 1`] = `
<div>
  <textarea
    class="flex min-h-[80px] w-full rounded-md border border-divider bg-white px-3 py-2 text-base ring-offset-white placeholder:text-letter-caption focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-primary focus-visible:ring-offset-1 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
    data-testid="Textarea-testid"
  />
</div>
`;
